src/api/app/components/notification_filter_component.rb

```diff
@@ -1,27 +1,37 @@
     super
 
+    @notifications = notifications
     @user = user
-    @projects_for_filter = projects_for_filter
-    @groups_for_filter = groups_for_filter
+    @projects_for_filter = ProjectsForFilterFinder.new.call
+    @groups_for_filter = GroupsForFilterFinder.new.call
     @count = notifications_count
     @selected_filter = selected_filter
   end
 
   # TODO: Turn this into a query object `NotificationsCounter` and pass this as a default value
   #       to a keyword argument `count` in the `initialize` method
   def notifications_count
-    notifications = User.session.notifications.for_web
-    counted_notifications = notifications.unread.group(:notifiable_type).count
-    counted_notifications['incoming_requests'] = notifications.unread.for_incoming_requests(User.session!).count
-    counted_notifications['outgoing_requests'] = notifications.unread.for_outgoing_requests(User.session!).count
-    counted_notifications['relationships_created'] = notifications.unread.for_relationships_created.count
-    counted_notifications['relationships_deleted'] = notifications.unread.for_relationships_deleted.count
-    counted_notifications['build_failures'] = notifications.unread.for_failed_builds.count
-    counted_notifications['reports'] = notifications.unread.for_reports.count
-    counted_notifications['workflow_runs'] = notifications.unread.for_workflow_runs.count
-    counted_notifications['appealed_decisions'] = notifications.unread.for_appealed_decisions.count
-    counted_notifications.merge!('unread' => User.session.unread_notifications)
+    counted_notifications = {}
+    counted_notifications['unread'] = @notifications.unread.count
+    counted_notifications['read'] = @notifications.read.count
+    counted_notifications['comments'] = @notifications.for_comments.count
+    counted_notifications['requests'] = @notifications.for_requests.count
+    counted_notifications['incoming_requests'] = @notifications.for_incoming_requests(User.session).count
+    counted_notifications['outgoing_requests'] = @notifications.for_outgoing_requests(User.session).count
+    counted_notifications['relationships_created'] = @notifications.for_relationships_created.count
+    counted_notifications['relationships_deleted'] = @notifications.for_relationships_deleted.count
+    counted_notifications['build_failures'] = @notifications.for_build_failures.count
+    counted_notifications['reports'] = @notifications.for_reports.count
+    counted_notifications['workflow_runs'] = @notifications.for_workflow_runs.count
+    counted_notifications['appealed_decisions'] = @notifications.for_appealed_decisions.count
+    @projects_for_filter.each do |project_name|
+      counted_notifications[project_name] = @notifications.for_project_name(project_name).count
+    end
+    @groups_for_filter.each do |group_title|
+      counted_notifications[group_title] = @notifications.for_group_title(group_title).count
+    end
```

Comment:
Nitpick: what if a group has the same name as a project? I invent some: group `vim` and project `vim`.

I won't block this PR for this hypothetical situation, but might have it in mind for the future.

Reply:
Good catch, fixed in https://github.com/openSUSE/open-build-service/pull/16183/commits/1ebedfdab8c4c82dd2d1bbe9ed3f197bbab892d4